botorch/posteriors/gpytorch.py

```diff
@@ -99,3 +100,41 @@ def variance(self) -> Tensor:
         if not self._is_mt:
             variance = variance.unsqueeze(-1)
         return variance
+
+
+def affine_transform(
+    posterior: GPyTorchPosterior, weights: Tensor, offset: float = 0.0
```

Comment:
When would we want to use a non-zero offset?

Reply:
We could use this for KG to pass in the (negative) current best, which could make the KG implementation somewhat more elegant.

Reply:
@danielrjiang @VilockLi